internal/trace/context.go

```diff
@@ -119,46 +123,52 @@ func createDummySubsegmentForXrayConverter(ctx context.Context, traceCtx TraceCo
 	return nil
 }
 
-// getDatadogTraceContextFromEvent extracts the Datadog trace context from an incoming Lambda event payload
-// and creates a dummy X-Ray subsegment containing this information
-func getDatadogTraceContextFromEvent(ctx context.Context, ev json.RawMessage) (TraceContext, bool) {
-	eh := eventWithHeaders{}
-
-	traceCtx := map[string]string{}
-
-	err := json.Unmarshal(ev, &eh)
-	if err != nil {
-		return traceCtx, false
-	}
+func getContext(context map[string]string) (Context, bool) {
+	tc := Context{}
 
-	lowercaseHeaders := map[string]string{}
-	for k, v := range eh.Headers {
-		lowercaseHeaders[strings.ToLower(k)] = v
+	traceID, ok := context[traceIDHeader]
+	if !ok {
+		return tc, false
 	}
 
-	traceID, ok := lowercaseHeaders[traceIDHeader]
+	parentID, ok := context[parentIDHeader]
 	if !ok {
-		return traceCtx, false
+		return tc, false
 	}
 
-	parentID, ok := lowercaseHeaders[parentIDHeader]
+	samplingPriority, ok := context[samplingPriorityHeader]
 	if !ok {
-		return traceCtx, false
+		return tc, false
 	}
 
-	samplingPriority, ok := lowercaseHeaders[samplingPriorityHeader]
-	if !ok {
-		return traceCtx, false
+	tc[samplingPriorityHeader] = samplingPriority
+	tc[traceIDHeader] = traceID
+	tc[parentIDHeader] = parentID
+
+	return tc, true
+}
+
+// getDatadogTraceContextFromHeaders extracts the Datadog trace context from an incoming Lambda event payload
+// and creates a dummy X-Ray subsegment containing this information.
+func getDatadogTraceContextFromHeaders(ctx context.Context, ev json.RawMessage) map[string]string {
```

Comment:
I don't agree with this rename -- I guess I don't really agree with the original name either. We're getting headers (which may or may not contain a trace context) from the event. It might also be worth calling out in the godoc that this is the DefaultTraceExtractor